### PR TITLE
[IOTDB-5125] Change default port and stop the process when port is already in use

### PR DESCRIPTION
**Pull request by Caideyipi**
### Port change
Changed the dn_schema_region_consensus_port from 50010 to 40030, because it conflicts with a port of hdfs
Related doc is: https://apache-iotdb.feishu.cn/docx/doxcnjzB2xd9sScIlkNYJUJA1zb
Wait until ip:port change is supported.

### Added check
Now the shell can detect whether the ports set are occupied, and print the processes occuping them.
Test pictures:
image
image
image
image

### Perfected the shell
Added some hint for users to correctly run the shell. 
See details in "File changes" column.

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/assembly/resources/sbin/start-datanode.bat [hunk 1]
```diff
@@ -89,7 +90,107 @@ IF EXIST "%IOTDB_CONF%\datanode-env.bat" (
...
+  for /f  "eol=# tokens=2 delims==" %%i in ('findstr /i "^dn_data_region_consensus_port"
+    %IOTDB_HOME%\conf\iotdb-datanode.properties') do (
+      set dn_data_region_consensus_port=%%i
+  )
+) ELSE (
+  echo "Can't find iotdb-datanode.properties, check the default ports"
+  set dn_rpc_port=6667
+  set dn_internal_port=9003
+  set dn_mpp_data_exchange_port=8777
+  set dn_schema_region_consensus_port=50010
```
Comment (CRZbulabula) [inline server/src/assembly/resources/sbin/start-datanode.bat:144]:
```suggestion
  set dn_schema_region_consensus_port=40030
```

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL and don't forget to update the related user docs

File: node-commons/src/assembly/resources/sbin/iotdb-common.sh [hunk 2]
```diff
@@ -150,6 +150,143 @@ checkAllConfigNodeVariables()
...
+    dn_schema_region_consensus_port=$(sed '/^dn_schema_region_consensus_port=/!d;s/.*=//' "${IOTDB_CONF}"/iotdb-datanode.properties)
+    dn_data_region_consensus_port=$(sed '/^dn_data_region_consensus_port=/!d;s/.*=//' "${IOTDB_CONF}"/iotdb-datanode.properties)
+  elif [ -f "$IOTDB_HOME/conf/iotdb-datanode.properties" ]; then
+    dn_rpc_port=$(sed '/^dn_rpc_port=/!d;s/.*=//' "${IOTDB_HOME}"/conf/iotdb-datanode.properties)
+    dn_internal_port=$(sed '/^dn_internal_port=/!d;s/.*=//' "${IOTDB_HOME}"/conf/iotdb-datanode.properties)
+    dn_mpp_data_exchange_port=$(sed '/^dn_mpp_data_exchange_port=/!d;s/.*=//' "${IOTDB_CONF}"/iotdb-datanode.properties)
+    dn_schema_region_consensus_port=$(sed '/^dn_schema_region_consensus_port=/!d;s/.*=//' "${IOTDB_CONF}"/iotdb-datanode.properties)
+    dn_data_region_consensus_port=$(sed '/^dn_data_region_consensus_port=/!d;s/.*=//' "${IOTDB_CONF}"/iotdb-datanode.properties)
+  else
+    echo "Cannot find iotdb-datanode.properties, check the default configuration"
```
Comment (Beyyes) [inline node-commons/src/assembly/resources/sbin/iotdb-common.sh:172]:
```suggestion
    echo "Warning: cannot find iotdb-datanode.properties, check the default configuration"
```

File: node-commons/src/assembly/resources/sbin/iotdb-common.sh [hunk 3]
```diff
@@ -150,6 +150,143 @@ checkAllConfigNodeVariables()
...
+      echo "The dn_schema_region_consensus_port" "$dn_schema_region_consensus_port" "is already occupied, PID:" "$PID"
+      occupied=true
+    fi
+    PID=$(netstat -anp 2>/dev/null | grep ":${dn_data_region_consensus_port} " | grep ' LISTEN ' | awk '{print $NF}' | sed "s|/.*||g")
+    if [ -n "$PID" ]; then
+      echo "The dn_data_region_consensus_port" "$dn_data_region_consensus_port" "is already occupied, PID:" "$PID"
+      occupied=true
+    fi
+  else
+    echo " Error: No necessary tool, stop ports checking"
```
Comment (Beyyes) [inline node-commons/src/assembly/resources/sbin/iotdb-common.sh:232]:
```suggestion
    echo " Error: No necessary tool to check whether given port is occupied, stop ports checking"
```

File: node-commons/src/assembly/resources/sbin/iotdb-common.sh [hunk 4]
```diff
@@ -150,6 +150,143 @@ checkAllConfigNodeVariables()
...
+      echo "The cn_internal_port" "$cn_internal_port" "is already occupied, PID:" "$PID"
+      occupied=true
+    fi
+    PID=$(netstat -anp 2>/dev/null | grep ":${cn_consensus_port} " | grep ' LISTEN ' | awk '{print $NF}' | sed "s|/.*||g")
+    if [ -n "$PID" ]; then
+      echo "The cn_consensus_port" "$cn_consensus_port" "is already occupied, PID:" "$PID"
+      occupied=true
+    fi
+  else
+    echo " Error: No necessary tool, stop ports checking"
```
Comment (Beyyes) [inline node-commons/src/assembly/resources/sbin/iotdb-common.sh:281]:
```suggestion
    echo " Error: No necessary tool to check whether given port is occupied, stop ports checking"
```

**Pull request merged**